### PR TITLE
Calculix-cgx 2.8

### PR DESCRIPTION
**Pull request by monofuel**
My first pull request, so sorry if I've made any mistakes.

Using the 2.7-r1 ebuild as a guide, I made a new ebuild for 2.8, and updated the patches. I've tested it on my system and haven't had any issues.

### CODE REVIEW

File: sci-libs/calculix-cgx/calculix-cgx-2.8.ebuild [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+RDEPEND="media-libs/mesa[nptl]
+	>=media-libs/freeglut-1.0"
+DEPEND="${RDEPEND}
+	doc? ( app-text/ghostscript-gpl )"
+
+S=${WORKDIR}/CalculiX/${MY_P}/src/
+
+src_unpack() {
+	default
+	cp "${DISTDIR}/${MY_P}.pdf" "${S}"
```
Comment (junghans) [inline sci-libs/calculix-cgx/calculix-cgx-2.8.ebuild]: I think, you only want to do that if `doc` is enabled and missing die. Better:
`use !doc || cp "${DISTDIR}/${MY_P}.pdf" "${S}" || die`

File: sci-libs/calculix-cgx/calculix-cgx-2.8.ebuild [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	#else
+	#	export PTHREAD=""
+	#fi
+}
+
+src_install () {
+	dobin cgx
+
+	if use doc; then
+		dodoc ${MY_P}.pdf
```
Comment (junghans) [inline sci-libs/calculix-cgx/calculix-cgx-2.8.ebuild]: Or simply use `${DISTDIR}` here.

File: sci-libs/calculix-cgx/calculix-cgx-2.8.ebuild [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	epatch "${FILESDIR}"/01_${MY_P}_Makefile_custom_cxx_flags.patch
+
+	#assuming this fix is still needed for 2.8
+	epatch "${FILESDIR}"/02_${MY_P}_menu_fix-freeglut_2.8.1.patch
+}
+
+src_configure () {
+	# Does not compile without -lpthread
+	#if use nptl; then
+	export PTHREAD="-lpthread"
```
Comment (junghans) [inline sci-libs/calculix-cgx/calculix-cgx-2.8.ebuild]: This line should be moved to `src_prepare` and drop `src_configure` completely.

File: sci-libs/calculix-cgx/files/01_cgx_2.8_Makefile_custom_cxx_flags.patch [hunk 4]
```diff
@@ -0,0 +1,76 @@
+diff -rupN CalculiX/cgx_2.8/src/Makefile CalculiX.patched/cgx_2.8/src/Makefile
```
Comment (junghans) [inline sci-libs/calculix-cgx/files/01_cgx_2.8_Makefile_custom_cxx_flags.patch]: A little header like described in http://dev.gentoo.org/~vapier/clean-patches would be nice.

**Comment by TheChymera** [comment]
package version superseded by current version in ::science

**Pull request closed without merging**